frontend/search/searchController.js

```diff
@@ -36,18 +36,22 @@
             searchCtrl.institutions = [];
         };
 
-        searchCtrl.search = function search() {
+        searchCtrl.search = function search(ev) {
             if (searchCtrl.search_keyword) {
-                searchCtrl.makeSearch(searchCtrl.search_keyword, 'institution');
+                searchCtrl.makeSearch(searchCtrl.search_keyword, 'institution').then(() => {
+                    if (Utils.isMobileScreen()) {
+                        searchCtrl.showSearchFromMobile(ev);
+                    }
+                });
                 refreshPreviousKeyword();
             }
         };
 
-        searchCtrl.goToInstitution = function goToInstitution(institutionId) {
+        searchCtrl.goToInstitution = function goToInstitution(institutionId, cameFromDialog) {
```

Comment:
create a function `gotoinstituition` with another name, to call in the dialog.
Inside do `md.cancel` and then call this `gotoinstitution`function.